roles/piwigo/tasks/main.yml

```diff
@@ -0,0 +1,40 @@
+---
+- name: Create Piwigo directories
+  file:
+    path: "{{ item }}"
+    state: directory
+  with_items:
+    - "{{ piwigo_data_directory }}/config"
+
+- name: Piwigo Mysql Docker Container
+  docker_container:
+    name: piwigo-mysql
+    image: mysql:5.7
+    pull: true
+    volumes:
+      - "{{ piwigo_data_directory }}/mysql:/var/lib/mysql:rw"
+    env:
+      MYSQL_DATABASE: "piwigo"
+      MYSQL_USER: "{{ piwigo_sql_user }}"
+      MYSQL_PASSWORD: "{{ piwigo_sql_password }}"
+      MYSQL_ROOT_PASSWORD: "{{ piwigo_sql_root_password }}"
+    restart_policy: unless-stopped
+
+- name: Piwigo Docker Container
+  docker_container:
+    name: piwigo
+    image: ghcr.io/linuxserver/piwigo
+    pull: true
+    volumes:
+      - "{{ piwigo_data_directory }}/config:/config:rw"
+      - "{{ piwigo_data_directory }}/pictures:/pictures"
```

Comment:
I used this volume mounting for piwigo docker container. It allowed the sync to pull directly from my samba photo share.
      - "{{ piwigo_photos_directory }}:/config/www/gallery/galleries/photos"